components/operator/internal/controllers/kubernetes/namespace_controller.go

```diff
@@ -72,19 +73,21 @@ func (r *NamespaceReconciler) Reconcile(ctx context.Context, request ctrl.Reques
 	logger := r.Log.With("name", instance.GetName())
 
 	logger.Debug(fmt.Sprintf("Updating Secret in namespace '%s'", instance.GetName()))
-	secret, err := r.secretSvc.GetBase(ctx)
-	if client.IgnoreNotFound(err) != nil {
-		logger.Error(err, "Getting base Secret failed")
-		return ctrl.Result{}, err
-	}
+	errs := []error{}
+	secrets, err := r.secretSvc.GetBase(ctx)
 	if err != nil {
-		logger.Debug(err, "Base Secret not found")
-		return ctrl.Result{}, nil
+		errs = append(errs, err)
 	}
-
-	if err := r.secretSvc.UpdateNamespace(ctx, logger, instance.GetName(), secret); err != nil {
-		return ctrl.Result{}, err
+	for _, secret := range secrets {
+		err = r.secretSvc.UpdateNamespace(ctx, logger, instance.GetName(), &secret)
+		if err != nil {
+			errs = append(errs, err)
+		}
 	}
 
-	return ctrl.Result{}, nil
+	err = goerrors.Join(errs...)
+	if err != nil {
+		logger.Error(err, "Updating base Secrets failed")
+	}
```

Comment:
I think there is no need for printing error details because if the `Reconcile` method returns an error then the controller will log an error, so to avoid error log duplication I would suggest removing these lines.

Reply:
removed